components/eamxx/src/physics/mam/readfiles/tracer_reader_utils.hpp

```diff
@@ -448,6 +448,34 @@ inline std::shared_ptr<AtmosphereInput> create_tracer_data_reader(
                                            true);
 }  // create_tracer_data_reader
 
+inline std::shared_ptr<AtmosphereInput> create_tracer_data_reader(
+    const std::shared_ptr<AbstractRemapper> &horiz_remapper,
+    const std::string &tracer_data_file,
+    const TracerData &tracer_data,
+    const std::string &extfrc_map_file
+    ) {
+  std::vector<Field> io_fields;
+  for(int i = 0; i < horiz_remapper->get_num_fields(); ++i) {
+    io_fields.push_back(horiz_remapper->get_src_field(i));
+  }
+  const auto io_grid = horiz_remapper->get_src_grid();
+
+  // NOTE: If we are using a vertical emission NC file with altitude instead of levels,
+  // we must rename this tag. This is only necessary when a map file is used.
+  if(tracer_data.file_type == VERT_EMISSION && extfrc_map_file != ""){
+    auto horiz_interp_src_grid =
+      io_grid->clone("tracer_horiz_interp_src_grid", true);
+  horiz_interp_src_grid->reset_field_tag_name(LEV, "altitude");
+  horiz_interp_src_grid->reset_field_tag_name(ILEV, "altitude_int");
+  return std::make_shared<AtmosphereInput>(tracer_data_file, horiz_interp_src_grid, io_fields,
+                                           true);
+  } else {
+    return std::make_shared<AtmosphereInput>(tracer_data_file, io_grid, io_fields,
+                                           true);
+  }
+
```

Comment:
He can't change the remapper src grid, since it's const. He'd have to (shallow) clone it first.

Reply:
I committed a cleaner version of this part of the code. I will clone only the source grid if the file type corresponds to a file with altitude, i.e., the `VERT_EMISSION` file type. I did not use a map because it would require additional changes in the interface code (I need to create a map). I deleted the overload of create_tracer_data_reader by passing the file type as input and setting the file type default to NONE.